tests/ftest.rb

```diff
@@ -0,0 +1,133 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+require 'yaml'
+require 'elasticsearch'
+
+LIB_DIR = File.expand_path("#{File.dirname(__FILE__)}/../lib")
+
+def es_client
+  Elasticsearch::Client.new(
+    host: 'http://localhost:9200',
+    user: 'elastic',
+    password: 'changeme',
+  )
+end
+
+def start_stack
+  # starts Mongo and Elasticsearch, adds data in Mongo
+  Dir.chdir(__dir__) {
+    puts('Start the stack')
+    system('make stop-stack')
+    system('make run-stack')
+    puts('Load mongo data')
+    system('make load-data')
+  }
+  puts('Wait for all kids to be up') # can do better (ping in a loop)
+  sleep 60
+end
+
+def prepare_es
+  puts('Wipe existing data')
+  system('curl -X DELETE http://localhost:9200/mongo --user elastic:changeme')
+end
+
+def set_auth
+  config_file = File.expand_path("#{File.dirname(__FILE__)}/../config/connectors.yml")
+
+  body = {
+    "name": 'mongo-connector',
+    "role_descriptors": {
+      "mongo-connector-role": {
+        "cluster": ['all'],
+        "index": [
+          {
+            "names": ['*'],
+            "privileges": ['all']
+          }
+        ]
+      }
+    }
+  }
+  response = es_client.perform_request('POST', '_security/api_key',
+                                       {}, body)
+
+  api_key = response.body['encoded']
+  # copy it into the config
+  config = YAML.load_file(config_file)
+  config['elasticsearch']['api_key'] = api_key.strip
+  File.open(config_file, 'w') { |file| file.write(config.to_yaml) }
+end
+
+def run_sync
+  # late import because the lib import the config when loaded
+  # and we change it in `set_auth`
+  $LOAD_PATH << LIB_DIR
+
+  Dir["#{LIB_DIR}/**/*.rb"].sort.each { |f|
+    next if f.include?('lib/app')
+    next if f.include?('connectors_async')
```

Comment:
Think we don't have this anymore

Reply:
What about https://github.com/elastic/connectors-ruby/blob/main/lib/connectors_async.rb ?